File: 프로그래머스/2/42578. 의상/의상.cpp
#include <string>
#include <vector>
#include <map>

using namespace std;

int solution(vector<vector<string>> clothes) {
    int answer = 1;
    
    map<string, int> M;
    for(int i=0; i<clothes.size(); i++) M[clothes[i][1]]++;
    
    for(auto iter=M.begin(); iter!=M.end(); iter++) answer *= iter-> second+1;
    
    return answer-1;
}

File: 프로그래머스/2/42578. 의상/의상.py
def solution(clothes):
    answer = 1
    
    clothe_cnt={}
    
    for (i,j) in clothes :
        if j in clothe_cnt :
            clothe_cnt[j] += 1
        else :
            clothe_cnt[j] = 1
    
    print(clothe_cnt)
    
    for i in clothe_cnt :
        answer *= (clothe_cnt[i]+1)
        
    return answer-1